### PR TITLE
CTL-895 dispatcher service abstraction leak

### PR DESCRIPTION
**Pull request by szogun1987**
Proposal of fix bug reported in [CTL-895](https://catelproject.atlassian.net/browse/CTL-895)

### CODE REVIEW

File: src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs [hunk 1]
```diff
@@ -56,12 +57,56 @@ public DispatcherService()
...
         }
 #endif
 
+#if NET
+        /// <summary>
+        /// Executes the specified delegate asynchronously with the specified arguments on the thread that the Dispatcher was created on.
+        /// </summary>
+        /// <param name="action">The action.</param>
+        /// <returns>The task representing the action.</returns>
+        public async Task InvokeAsync(Action action)
```
Comment (GeertvanHorrik) [inline src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs]: While we are at it now anyway, please refactor into this:
```
public async InvokeAsync(Action action)
{ 
    var dispatcher = CurrentDispatcher; 
 
#if NET40 
    return DispatcherExtensions.InvokeAsync(dispatcher, action);  
#else  
    return dispatcher.InvokeAsync(action); 
#endif  
} 
```

File: src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs [hunk 2]
```diff
@@ -56,12 +57,56 @@ public DispatcherService()
...
+        /// <param name="action">The action.</param>
+        /// <returns>The task representing the action.</returns>
+        public Task InvokeAsync(Action action)
+        {
+            var dispatcher = CurrentDispatcher;
+
+#if NET40
+            return DispatcherExtensions.InvokeAsync(dispatcher, action);
+#else
+            return dispatcher.InvokeAsync(action).Task;
```
Comment (szogun1987) [inline src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs]: I haven't test it. Maybe 2 separate InvokeAsync methods should be added (one for .Net 4.0 one for 4.5+)

Comment (GeertvanHorrik) [inline src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs]: I prefer to keep 1 method, I think this will work fine.

**Comment by GeertvanHorrik** [comment]
If you want, you can add yourself to the CONTRIBUTORS file. Also please add this to history.txt, then I can merge it.

Thanks for the PR!

**Comment by szogun1987** [comment]
I have started mine branch from master instead of develop. So I have merge it to mine one. Files updated please ensure that everything is OK.

File: doc/history.txt [hunk 3]
```diff
@@ -53,6 +53,8 @@ we will remove the features in the next major release.
 
 Added/fixed:
 ============
+(*) CTL-895 Removing CurrentDispatcher property from IDispatcherService interface. Adding methods to IDispatcher
```
Comment (GeertvanHorrik) [inline doc/history.txt]: We use ordering in the history.txt, but I can change this one once this has been accepted.

**Comment by GeertvanHorrik** [comment]
Thanks a lot for the PR!

**Pull request merged**